mlflow/system_metrics/system_metrics_monitor.py

```diff
@@ -0,0 +1,132 @@
+"""Class for monitoring system stats."""
+
+import logging
+import threading
+
+from mlflow.environment_variables import (
+    MLFLOW_SYSTEM_METRICS_SAMPLES_BEFORE_LOGGING,
+    MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL,
+)
+from mlflow.system_metrics.metrics.cpu_monitor import CPUMonitor
+from mlflow.system_metrics.metrics.disk_monitor import DiskMonitor
+from mlflow.system_metrics.metrics.gpu_monitor import GPUMonitor
+from mlflow.system_metrics.metrics.network_monitor import NetworkMonitor
+
+_logger = logging.getLogger(__name__)
+
+
+class SystemMetricsMonitor:
+    """Class for monitoring system stats.
+
+    This class is used for pulling system metrics and logging them to MLflow. Calling `start()` will
+    spawn a thread that logs system metrics periodically. Calling `finish()` will stop the thread.
+    Logging is done on a different frequency from pulling metrics, so that the metrics are
+    aggregated over the period. Users can change the logging frequency by setting
+    `$MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL` and `$MLFLOW_SYSTEM_METRICS_SAMPLES_BEFORE_LOGGING`
+    environment variables.
+
+    Args:
+        run_id: string, the MLflow run ID.
+        sampling_interval: float, default to 10. The interval (in seconds) at which to pull system
+            metrics.
+        samples_before_logging: int, default to 1. The number of samples to aggregate before
+            logging.
+    """
+
+    def __init__(self, run_id, sampling_interval=10, samples_before_logging=1):
+        from mlflow.utils.autologging_utils import BatchMetricsLogger
+
+        # Instantiate default monitors.
+        self.monitors = [CPUMonitor(), DiskMonitor(), NetworkMonitor()]
+        gpu_monitor = GPUMonitor()
+        if gpu_monitor:
+            self.monitors.append(gpu_monitor)
+        self.sampling_interval = MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL.get() or sampling_interval
+        self.samples_before_logging = (
+            MLFLOW_SYSTEM_METRICS_SAMPLES_BEFORE_LOGGING.get() or samples_before_logging
+        )
+
+        if isinstance(self.sampling_interval, str):
+            self.sampling_interval = float(self.sampling_interval)
+        if isinstance(self.samples_before_logging, str):
+            self.samples_before_logging = int(self.samples_before_logging)
+
+        self._run_id = run_id
+        self.mlflow_logger = BatchMetricsLogger(self._run_id)
+        self._shutdown_event = threading.Event()
+        self._process = None
+
+    def start(self):
+        """Start monitoring system metrics."""
+        try:
+            self._process = threading.Thread(
+                target=self.monitor,
+                daemon=True,
+                name="SystemMetricsMonitor",
+            )
+            self._process.start()
+            _logger.info("MLflow: started monitoring system metrics.")
```

Comment:
```suggestion
            _logger.info("started monitoring system metrics.")
```

Do we need `MLflow: `? I think the logging message is prefixed with the module name. `MLflow: ` seems redundant to me.

Reply:
I am mostly thinking from the GCP logging UI. There the file path is shown in a different field, having `MLflow: ` as a prefix won't be bothering, and it's clear that this system metrics logging is from MLflow.

Reply:
I agree on this if we do the same thing in other places.

Reply:
I cannot remember where I see that haha, I will delete it.